contracts/fuses/universal_token_swapper/UniversalTokenSwapperEthFuse.sol

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.26;
+
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {IERC20Metadata} from "@openzeppelin/contracts/interfaces/IERC20Metadata.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IFuseCommon} from "../IFuseCommon.sol";
+import {PlasmaVaultConfigLib} from "../../libraries/PlasmaVaultConfigLib.sol";
+import {IPriceOracleMiddleware} from "../../price_oracle/IPriceOracleMiddleware.sol";
+import {PlasmaVaultLib} from "../../libraries/PlasmaVaultLib.sol";
+import {IporMath} from "../../libraries/math/IporMath.sol";
+import {SwapExecutorEth, SwapExecutorEthData} from "./SwapExecutorEth.sol";
+
+/// @notice Data structure used for executing a swap operation.
+/// @param  targets - The array of addresses to which the call will be made.
+/// @param  data - Data to be executed on the targets.
+struct UniversalTokenSwapperEthData {
+    address[] targets;
+    bytes[] callData;
+    uint256[] ethAmounts;
+    address[] dustToCheck;
+}
+
+/// @notice Data structure used for entering a swap operation.
+/// @param  tokenIn - The token that is to be transferred from the plasmaVault to the swapExecutor.
+/// @param  tokenOut - The token that will be returned to the plasmaVault after the operation is completed.
+/// @param  amountIn - The amount that needs to be transferred to the swapExecutor for executing swaps.
+/// @param  data - A set of data required to execute token swaps
+struct UniversalTokenSwapperEthEnterData {
+    address tokenIn;
+    address tokenOut;
+    uint256 amountIn;
+    UniversalTokenSwapperEthData data;
+}
+
+struct Balances {
+    uint256 tokenInBalanceBefore;
+    uint256 tokenOutBalanceBefore;
+    uint256 tokenInBalanceAfter;
+    uint256 tokenOutBalanceAfter;
+}
+
+/// @title This contract is designed to execute every swap operation and check the slippage on any DEX.
+contract UniversalTokenSwapperEthFuse is IFuseCommon {
+    using SafeERC20 for ERC20;
+
+    event UniversalTokenSwapperFuseEnter(address version, address asset, uint256 amount);
```

Comment:
Event never used, please use event in enter method. Analogically do fix in UniversalTokenSwapperFuse